tools/linker/MonoTouch.Tuner/PreserveSmartEnumConversionsSubStep.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright 2017 Xamarin Inc.
+
+using System;
+using System.Collections.Generic;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Linker;
+using Mono.Tuner;
+using MonoTouch.Tuner;
+
+namespace Xamarin.Linker.Steps
+{
+	public class PreserveSmartEnumConversionsSubStep : ExceptionalSubStep
+	{
+		Dictionary<TypeDefinition, Tuple<MethodDefinition, MethodDefinition>> cache;
+		protected override string Name { get; } = "Smart Enum Conversion Preserver";
+		protected override int ErrorCode { get; } = 2100;
+
+		public override SubStepTargets Targets {
+			get {
+				return SubStepTargets.Method | SubStepTargets.Property;
+			}
+		}
+
+		public override bool IsActiveFor (AssemblyDefinition assembly)
+		{
+			// we need to process all assemblies, because the functions we want to
+			// preserve are not necessarily in the assembly we're processing.
+			return true;
```

Comment:
I initially copy-pasted code that returned true depending on whether the assembly was linked or not, and in with that condition it makes sense to process every assembly.

It didn't occur to me that I could apply other criteria.